design/specs/list-data-type.md

```diff
@@ -0,0 +1,194 @@
+---
+title: List Data Type
+description: 
+published: false
+date: 2022-04-10 18:02:53
+tags: 
+editor: markdown
+dateCreated: 2022-04-06 18:02:58
+---
+
+# Context
+
+This document specifies the design of List data type within the Mathesar application. List data type can include one or more items that are in the same basic data type (e.g., Number), separated by a comma delimiter if more than one item is present. Mathesar aims to support all basic data types in List format. Users will be able to view data that is currently in Mathesar List data type, edit and filter List data, create new columns in List data type, and convert existing data into List data type. 
+
+The goal of the design in general is to help non-technical understand List data type and maintain List data integrity. Aggregating data into Lists via mapping tables is not beyond the scope of this spec and should be designed in visually different ways to communicate the affordance.
+
```

Comment:
To me, the mapping tables and aggregated Lists for Views feel like a different concept, and it might be beneficial to distinguish it in significant ways, possibly even with a different name. 

It's quite possible I haven't understood the connection between a "native" List in a single cell vs. an aggregated List. We can discuss.

Reply:
When it comes to the view, the data contained within an aggregated column will be treated as a list, much like a list in a table. However, the editing interactions for each must be distinct. A view, aside from having a query behind it, should look like a table.

Reply:
Thanks for the clarification! These bullet points are in line with what I'm thinking. The editing flow described in 1.3 can accomplish all these except deleting multiple list items at once – I treated editing a cell with List similar to a regular cell, just with pronounced List structure indication to help users – would love to know your thought.